### PR TITLE
Adds PHPUnit tests for gutenberg_register_fonts_from_theme_json().

### PR DESCRIPTION
**Pull request by hellofromtonya**
## What?
Adding PHPUnit tests for `gutenberg_register_fonts_from_theme_json()`.

## Why?
- To provide code coverage specifically for this function as it is critical for the automatic registration and enqueuing of the theme's `theme.json` and style variation fonts and variations.
- To set a baseline of automated tests before improvement happens for this function.

## How?

Adds a theme with fonts defined in its `theme.json` and a mixture of scenarios in its style variations.

Adds a test class for this function with the following coverage:

- Skips when theme does not have fonts defined.
- Registers and enqueues style variations font family and its variations.
- Registers and enqueued all of the fonts from the theme's `theme.json` and style variations.
- No duplicate font families or variations are not registered. How? The `variations-duplicate-fonts` style variation includes with duplicate DM Sans variations.
- Converts `src` `file:./` into a URL to the font's asset file.
- Converts json style properties into CSS kebab-style, i.e. `fontFamily` into `font-family`.
- Skips re-registering and enqueuing all fonts that are already registered. How? Uses the 'origin' property.
- Skips fonts where the `fontFace` is not defined.

## Missing test coverage

Testing the variation font-family handle logic is missing from this PR. Why? There's possibly a flaw in the logic that needs investigation. This makes it out-of-scope for this PR.

## Testing Instructions
CI is green.

**Manual**

```
npm run test:unit:php -- --filter Tests_Fonts_GutenbergRegisterFontsFromThemeJson
```

### CODE REVIEW

File: phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+ * @package WordPress
+ * @subpackage Fonts API
+ *
+ * @since X.X.X
+ * @group fontsapi
+ * @covers ::gutenberg_register_fonts_from_theme_json
+ */
+class Tests_Fonts_GutenbergRegisterFontsFromThemeJson extends WP_Fonts_TestCase {
+	const FONTS_THEME    = 'fonts-block-theme';
+	const FONT_FAMILIIES = array(
```
Comment (anton-vlasenko) [inline phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php]:
```suggestion
	const FONT_FAMILIES = array(
```

File: phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+			$property->setAccessible( true );
+			$data = $property->getValue( $theme_json_object );
+
+			// Loop through the fonts to find the expected font-family to modify.
+			foreach ( $data['settings']['typography']['fontFamilies']['theme'] as $index => $definitions ) {
+				if ( $expected_font_family !== $definitions['slug'] ) {
+					continue;
+				}
+
+				// Remove the 'fontFace' element, which removes the font's varations.
```
Comment (anton-vlasenko) [inline phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php]:
```suggestion
				// Remove the 'fontFace' element, which removes the font's variations.
```

File: phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+
+require_once __DIR__ . '/wp-fonts-testcase.php';
+
+/**
+ * Test gutenberg_register_fonts_from_theme_json().
+ *
+ * @package WordPress
+ * @subpackage Fonts API
+ *
+ * @since X.X.X
```
Comment (anton-vlasenko) [inline phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php:11]: Should this be more specific?
```suggestion
 * @since 15.7.0
```

**Review by anton-vlasenko — COMMENTED** [review]
I have added some minor suggestions. Thanks!

File: phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php (hunk 3 above)
Comment (hellofromtonya) [reply phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php:11]: The `X.X.X` is a placeholder for when it's introduced into WP Core. The plan is to introduce the function and this test class into Core, hopefully in 6.3. So the `X.X.X`s in the API would all be updated to `6.3.0` at that time.

**Comment by hellofromtonya** [comment]
Thank you for reviewing @anton-vlasenko 😃  I've committed both of your typo fixes - thank you - as well as left the reasoning for why `X.X.X` is being used in the API and its tests. Would greatly appreciate another review from you when you have time.

File: phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php (hunk 1 above)
Comment (hellofromtonya) [reply phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php]: Suggestion committed in https://github.com/WordPress/gutenberg/pull/49877/commits/aeedd6d8091f41d1f53d46aacc241b52fe636608. Thank you!

File: phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php (hunk 2 above)
Comment (hellofromtonya) [reply phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php]: Suggestion committed in https://github.com/WordPress/gutenberg/pull/49877/commits/aeedd6d8091f41d1f53d46aacc241b52fe636608. Thank you!

**Review by anton-vlasenko — APPROVED** [review]
Thank you for the quick reply, @hellofromtonya.
I was not able to find anything except the typos I mentioned.
The code looks good to me 👍
Therefore, I approved the PR.

File: phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php (hunk 3 above)
Comment (anton-vlasenko) [reply phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php:11]: Thank you for the explanation.
I have not encountered this notation before.
Now I know that such a practice exists.

**Pull request merged**